test/spsc_queue/test_spsc_queue.ml

```diff
@@ -31,7 +31,7 @@ let test_parallel () =
   let producer =
     Domain.spawn (fun () ->
         for i = 1 to count do
-          while not (push_not_full q i) do
+          while not (push_not_full q (Float.of_int i)) do
```

Comment:
I think we should either functorize the tests or maybe use some other tricks to test both the safe and unsafe versions.

Reply:
I pushed a solution to run all the tests with both the safe and the unsafe implementations ! Thanks for your review.